enigma/src/main/java/cuchaz/enigma/analysis/index/JarIndex.java

```diff
@@ -84,7 +84,12 @@ public void indexJar(Set<String> classNames, ClassProvider classProvider, Progre
 		progress.step(2, I18n.translate("progress.jar.indexing.references"));
 
 		for (String className : classNames) {
-			classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			try {
+				classProvider.get(className).accept(new IndexReferenceVisitor(this, entryIndex, inheritanceIndex, Enigma.ASM_VERSION));
+			} catch (Exception e) {
+				System.err.println("Exception while indexing class: " + className + ":");
+				e.printStackTrace();
```

Comment:
This should throw an new RuntimeException with a message containing the class name.

We dont want to contiune when in a bad state.